internal/jobcontainers/oci.go

```diff
@@ -0,0 +1,77 @@
+package jobcontainers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Microsoft/hcsshim/internal/jobobject"
+	"github.com/Microsoft/hcsshim/internal/processorinfo"
+
+	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/Microsoft/hcsshim/internal/oci"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/sirupsen/logrus"
+)
+
+// This file contains helpers for converting parts of the oci spec to useful
+// structures/limits to be applied to a job object.
+
+// Oci spec to job object limit information. Will do any conversions to job object specific values from
+// their respective OCI representations. E.g. we convert CPU count into the correct job object cpu
+// rate value internally.
+func specToLimits(ctx context.Context, s *specs.Spec) (*jobobject.JobLimits, error) {
```

Comment:
Assuming most of this code is identical to what we use in hcsdoc_wcow.go, can we refactor so that it isn't duplicated?

Reply:
Sure, makes sense